elastic/kibana
# [Logs+] Fix landing page log data check and redirect

## :memo: Summary

This registers a log data check for use on the serverless deployment's landing page.

closes https://github.com/elastic/kibana/issues/160205

## :female_detective: Review notes

- Before this, the only place that registers that data check was the `infra` plugin, which will not be enabled in serverless deployments.
- We recognize that these immediate redirects might be confusing to the user. We're already discussing improvements to the landing page as follow-ups.
- The redirects on the landing page were previously implemented right in the component function body. I changed it to an effect to adhere to good practices.

## Testing notes

- Disable the `infra` plugin in the observability serverless config by setting `xpack.infra.enabled: false`.
- Deploy an observability serverless Kibana connected to a cluster that doesn't contain any log indices.
- Check that the landing page redirects to the on-boarding page.
- Create a data stream that matches `logs-*-*` and ingest at least one log entry document in it.
- Check that the landing page redirects to Discover with the `log-explorer` profile.
